src/vmm/src/lib.rs

```diff
@@ -522,13 +531,17 @@ impl Vmm {
         let device_states = self.mmio_device_manager.save();
 
         let memory_state = self.guest_memory().describe();
+        #[cfg(target_arch = "x86_64")]
+        let acpi_dev_state = self.acpi_device_manager.save();
 
         Ok(MicrovmState {
             vm_info: vm_info.clone(),
             memory_state,
             vm_state,
             vcpu_states,
             device_states,
+            #[cfg(target_arch = "x86_64")]
+            acpi_dev_state,
```

Comment:
As I recall when we change the state of the snapshot we should increase the snapshot version.

Reply:
good catch!

Reply:
We should think about some sort of test for this. Maybe A/B test with "before PR" binary and "after PR" binary where we just try to restore snap from one in another if reported snap versions are same?
Obviously this should not be a part of this PR.